media-proxy/include/mesh/conn_rdma.h

```diff
@@ -0,0 +1,128 @@
+#ifndef CONN_RDMA_H
+#define CONN_RDMA_H
+
+#include "logger.h"
+#include "mesh/conn.h"
+#include "concurrency.h"
+#include "libfabric_ep.h"
+#include "libfabric_mr.h"
+#include "libfabric_cq.h"
+#include "libfabric_dev.h"
+#include "mcm_dp.h"
+#include <mutex>
+#include <cstring>
+#include <cstddef>
+#include <atomic>
+#include <queue>
+
+#ifndef RDMA_DEFAULT_TIMEOUT
+#define RDMA_DEFAULT_TIMEOUT 1
+#endif
+#ifndef MAX_BUFFER_SIZE
+#define MAX_BUFFER_SIZE 1 << 30
+#endif
+#ifndef CQ_BATCH_SIZE
+#define CQ_BATCH_SIZE 64
+#endif
+#ifndef PAGE_SIZE
+#define PAGE_SIZE 4096
+#endif
+
+namespace mesh {
+
+namespace connection {
+
+/**
+ * Rdma
+ *
+ * Base class for RDMA connections, derived from the `Connection` class.
+ * Provides common RDMA-related functionality and acts as a foundation
+ * for specialized RDMA Tx and Rx classes.
+ */
+class Rdma : public Connection {
+  public:
+    Rdma();
+    virtual ~Rdma();
+
+  //Used only for Unit tests, provides access to protected members
+  #ifdef UNIT_TESTS_ENABLED
+      size_t get_buffer_queue_size() const { return buffer_queue.size(); }
+      bool is_buffer_queue_empty() const { return buffer_queue.empty(); }
+      Kind get_kind() const { return _kind; }
+      void* get_buffer_block() const { return buffer_block; }
+  #endif
+
+    // Queue synchronization
+    void init_buf_available();
+    void notify_buf_available();
+    void wait_buf_available();
+
+  protected:
+    // Configure the RDMA session
+    virtual Result configure(context::Context& ctx, const mcm_conn_param& request,
+                             const std::string& dev_port, libfabric_ctx *& dev_handle, Kind kind,
```

Comment:
Please set ```Kind kind``` inside RdmaRx/RdmaTx constructor, you dont have to pass it through ```Configure()``` method

Reply:
fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/286/commits/1e7f66891946f5a904408fb568ca119b7e04eaa0